scipy/stats/_mannwhitneyu.py

```diff
@@ -0,0 +1,367 @@
+# Using `nametuple` for now to pass existing mannwhitneyu tests without
+# modification
+MannwhitneyuResult = namedtuple('MannwhitneyuResult', ('statistic', 'pvalue'))
+
+
+def mannwhitneyu2(x, y, use_continuity=True, alternative="two-sided",
+                  axis=0, method="asymptotic"):
+    r'''Perform the Mann-Whitney U rank test on two independent samples.
+
+    The Mann-Whitney U test is a nonparametric test of the null hypothesis
+    that the underlying distribution of sample `x` is neither stochastically
+    greater nor stochastically less than the uderlying distribution of sample
+    `y`.
+
+    Parameters
+    ----------
+    x, y : array-like
+        N-d arrays of samples. The arrays must be broadcastable except along
+        the dimension given by `axis`.
+    use_continuity : bool, optional
+            Whether a continuity correction (1/2) should be applied.
+            Default is True when `method` is ``'asymptotic'``; has no effect
+            otherwise.
+    alternative : {'two-sided', 'less', 'greater'}, optional
+        Defines the alternative hypothesis.
+        The following options are available:
+
+        * 'two-sided' (default)
+        * 'less': one-sided
+        * 'greater': one-sided
+
+    axis : int, optional
+        Axis along which to perform the test. Default is 0.
+    method : {'asymptotic', 'exact'}, optional
+        Selects the method used to calculate the *p*-value. The following
+        options are available.
+
+        * ``'asymptotic'``: (default) compares the standardized test statistic
+          against the normal distribution, correcting for ties.
+        * ``'exact'``: computes the exact *p*-value by comparing the observed
+          :math:`U` statistic against the exact distribution of the :math:`U`
+          statistic under the null hypothesis. No correction is made for ties.
+
+    Returns
+    -------
+    res : MannwhitneyuResult
+        An object containing attributes:
+
+        statistic : float
+            The Mann-Whitney U statistic corresponding with sample `x`. See
+            Notes for the test statistic corresponding with sample `y`.
+        pvalue : float
+            The associated *p*-value for the chosen `alternative`.
+
+    Notes
+    -----
+    If ``U1`` is the statistic corresponding with sample `x`, then the
+    statistic corresponding with sample `y` is
+    `U2 = `x.shape[axis] * y.shape[axis] - U1``.
+
+    `mannwhitneyu2` is for independent samples. For related samples,
+    consider `scipy.stats.wilcoxon`.
+
+    `method` ``'exact'`` is recommended when there are no ties and when either
+    sample size is less than 8 [1]_. The implementation follows the recurrence
+    relation originally proposed in [1]_ as it is described in [3]_.
+    Note that the exact method is *not* corrected for ties, but
+    `mannwhitneyu2` will not raise errors or warnings if there are ties in the
+    data.
+
+    See Also
+    --------
```

Comment:
please add scipy.stats.ttest_ind

we should also point out somewhere that MWU is the non-parametric version of the t-test for independent samples

it could also be interesting to compare the results in the example to that of a t-test

Reply:
Sure, I'll include `scipy.stats.ttest_ind` as see also. 

> we should also point out somewhere that MWU is the non-parametric version of the t-test for independent samples

I see. Sometimes the accurate wording of these things is subtle. Do you know of a reference, or can you suggest a wording? Just:

> The Mann-Whitney U test is a non-parametric version of the t-test for independent samples.

?

I'll try comparing the two.

Reply:
Done in next push.

Reply:
Done (next push)

Reply:
I just noted that there are various ways to specify the null hypothesis of MWU. one of them is related to the comparison of the means. sorry about the confusion. I will share a screenshot of a book by email. see also https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2857732/pdf/nihms-185373.pdf

Reply:
Thanks. I commented more above, but the reason I didn't write it in terms of the means is because that requires other assumptions about the distributions. We could consider writing those assumptions and the corresponding null hypothesis, but originally I wanted to keep it general.